### PR TITLE
Fix dist missing issue for npm package installing

### PR DESCRIPTION
**Pull request by yma88**

### CODE REVIEW

File: charon/constants.py [hunk 1]
```diff
@@ -173,3 +173,5 @@
 PROD_INFO_SUFFIX = ".prodinfo"
 MANIFEST_SUFFIX = ".txt"
 DEFAULT_ERRORS_LOG = "errors.log"
+
+NRRC_REGISTRY = "npm.registry.redhat.com"
```
Comment (ligangty) [inline charon/constants.py:177]: I think we should leave this out as a configuration item below each "target", as it is not same in different environments. We can set this as DEFAULT_NRRC_REGISTRY with a default value(for example "localhost" ) if it is not specified in configuration,

File: charon/utils/archive.py [hunk 2]
```diff
@@ -81,12 +87,33 @@ def _get_tgz_name(path: str):
     return ""
 
 
-def __parse_npm_package_version_paths(path: str) -> list:
+def _del_none(d):
```
Comment (ligangty) [inline charon/utils/archive.py:90]: I remember there is a same func in npm.py, if so, please remove it and reuse this one.

File: charon/utils/archive.py [hunk 3]
```diff
@@ -81,12 +87,33 @@ def _get_tgz_name(path: str):
...
+    return d
+
+
+def __write_npm_version_dist(path: str, version_meta_extract_path: str, version_data: dict,
+                             tgz_relative_path: str):
+    tarball_link = "".join(["https://", NRRC_REGISTRY, "/", tgz_relative_path])
+    shasum = digest(path, HashType.SHA1)
+    dist = dict()
+    dist["tarball"] = tarball_link
+    dist["shasum"] = shasum
```
Comment (ligangty) [inline charon/utils/archive.py:105]: There is also a "integrity" item in this dist section, which has "sha512-xxxxxx" value.

**Review by ligangty — COMMENTED** [review]
Something needs to change as comments. And I'd suggest to add a unit test to verify this case.

File: charon/constants.py (hunk 1 above)
Comment (yma88) [reply charon/constants.py:177]: @ligangty If for multi targets uploading, and configuring different registries with the targets, it has to handle the different metadata content-generating for different target/env, I want to confirm there will be any such case we will encounter in prod/stage?

Comment (ligangty) [reply charon/constants.py:177]: Yes there will be such case. We will not prevent user to upload one tarball to both prod/stage in one time.

File: charon/pkgs/npm.py [hunk 4]
```diff
@@ -86,48 +87,47 @@ def handle_npm_uploading(
...
+        target_dir, valid_paths, package_metadata = _scan_metadata_paths_from_archive(
+            tarball_path, registry__, prod=product, dir__=dir_
+        )
+        if not os.path.isdir(target_dir):
+            logger.error("Error: the extracted target_dir path %s does not exist.", target_dir)
+            sys.exit(1)
+        valid_dirs = __get_path_tree(valid_paths, target_dir)
+
+        # main_target = targets[0]
+        client = S3Client(aws_profile=aws_profile, dry_run=dry_run)
```
Comment (ligangty) [inline charon/pkgs/npm.py:103]: To save performance, we don't need to initiate this client multi times for different targets.

**Review by ligangty — COMMENTED** [review]
Generally looks good except a small performance point.

File: charon/pkgs/npm.py (hunk 4 above)
Comment (yma88) [reply charon/pkgs/npm.py]: @ligangty, fixed, thanks!

**Review by ligangty — APPROVED** [review]
LGTM

**Pull request merged**